src/main/java/io/jenkins/plugins/sprp/YamlRootAction.java

```diff
@@ -0,0 +1,75 @@
+package io.jenkins.plugins.sprp;
+
+import hudson.Extension;
+import hudson.model.UnprotectedRootAction;
+import hudson.security.csrf.CrumbExclusion;
+import jenkins.model.Jenkins;
+import org.jenkinsci.plugins.workflow.multibranch.WorkflowMultiBranchProject;
+import org.kohsuke.stapler.StaplerRequest;
+import org.kohsuke.stapler.StaplerResponse;
+
+import javax.servlet.FilterChain;
+import javax.servlet.ServletException;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import java.io.IOException;
+import java.util.List;
+import java.util.logging.Logger;
+
+/**
+ * @author Honza Brázdil jbrazdil@redhat.com
+ */
+@Extension
+public class YamlRootAction implements UnprotectedRootAction {
+
+    static final String URL = "yamlhook";
+    static final String BUILD_CONFIGURATION_MODE = "by Jenkinsfile.yaml";
+
+    private static final Logger LOGGER = Logger.getLogger(YamlRootAction.class.getName());
+
+    public String getIconFileName() {
+        return null;
+    }
+
+    public String getDisplayName() {
+        return null;
+    }
+
+    public String getUrlName() {
+        return URL;
+    }
+
+    public void doIndex(StaplerRequest req,
+                        StaplerResponse resp) {
+        List<WorkflowMultiBranchProject> projects = Jenkins.get().getItems(WorkflowMultiBranchProject.class);
```

Comment:
It won't work as you expect. Even if the action itself is not protected, this method should have permission checks inside

Reply:
Can you suggest something using which I can do the same thing and permissions are taken care of.